### DISCUSSION TITLE
Feature request: integrate Q Library for unrivaled fast pitch detection

### DISCUSSION THREAD

**Discussion by mkoenings** [pschatzmann/arduino-audio-tools#363] [Ideas]
It would be awsome to be able to use this audio DSP library: 
Cycfi [Q library](https://github.com/cycfi/q)
It contains is the fastest pitch detection I know. I have used it successfully with a Daisy board,  thanks to the great work found here:
[Daisy Q](https://github.com/recursinging/daisy_q)
It can be used to create guitar synth apps, guitar to midi, , tuners, etc. It does not use much CPU power,which is a big plus.
Thanks!

**Comment by pschatzmann** [comment]
What is the major difference between Pitch Detection and FFT ?
Did you do some measurements on the ESP32 or other microcontrollers ?
You could also try to use Faust which has a Pitch Tracker..

**Comment by mkoenings** [comment]
The author came up with a concept thar works on a bitstream level, ( bit, not byte :-) ).
He explains that in detail [here](https://www.cycfi.com/2018/03/fast-and-efficient-pitch-detection-bitstream-autocorrelation/). He needed that for his very sophisticated hex pickup guitar ([Thor](https://www.cycfi.com/2021/12/we-are-ready/)).
The main differenece is the speed, and the small footprint in terms of CPU, which is a bit of a problem with FFT. The Thor guitar uses 6 signals at the same time, whic are processed in "real"-time on one MCU, check out the sound samples if you find the time, its pretty amazing.
Thanks!
